eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/HtlcTlv.scala

```diff
@@ -34,9 +34,15 @@ object UpdateAddHtlcTlv {
   /** Blinding ephemeral public key that should be used to derive shared secrets when using route blinding. */
   case class BlindingPoint(publicKey: PublicKey) extends UpdateAddHtlcTlv
 
+  case class Endorsement(level: Int) extends UpdateAddHtlcTlv
+
   private val blindingPoint: Codec[BlindingPoint] = (("length" | constant(hex"21")) :: ("blinding" | publicKey)).as[BlindingPoint]
 
-  val addHtlcTlvCodec: Codec[TlvStream[UpdateAddHtlcTlv]] = tlvStream(discriminated[UpdateAddHtlcTlv].by(varint).typecase(UInt64(0), blindingPoint))
+  private val endorsement: Codec[Endorsement] = tlvField(uint8.narrow[Endorsement](n => if (n >= 8) Attempt.failure(Err(s"invalid endorsement level: $n")) else Attempt.successful(Endorsement(n)), _.level))
+
+  val addHtlcTlvCodec: Codec[TlvStream[UpdateAddHtlcTlv]] = tlvStream(discriminated[UpdateAddHtlcTlv].by(varint)
+    .typecase(UInt64(0), blindingPoint)
+    .typecase(UInt64(106823), endorsement))
```

Comment:
Can you create a dedicated PR that just implements bLIP 4 and relays the incoming endorsement value? Adding our own reputation algorithm to compute it is a second (logical) step *after* adding that and is more easily reviewed separately.

Reply:
https://github.com/ACINQ/eclair/pull/2884